gensim/models/fasttext.py

```diff
@@ -0,0 +1,394 @@
+        # These two gives same result
+        for w, v in self.wv.vocab.items():
+            self.wv.syn0[v.index] = self.word_vec(w)
+
+    def word_vec(self, word, use_norm=False):
+        if word in self.wv.vocab:
+            if use_norm:
+                return self.wv.syn0norm[self.wv.vocab[word].index]
+            else:
+                return self.wv.syn0[self.wv.vocab[word].index]
+        else:
+            word_vec = np.zeros(self.wv.syn0_all.shape[1])
+            ngrams = Ft_Wrapper.compute_ngrams(word, self.min_n, self.max_n)
+            ngrams = [ng for ng in ngrams if ng in self.wv.ngrams]
+            if use_norm:
+                ngram_weights = self.wv.syn0_all_norm
+            else:
+                ngram_weights = self.wv.syn0_all
+            for ngram in ngrams:
+                word_vec += ngram_weights[self.wv.ngrams[ngram]]
+            if word_vec.any():
+                return word_vec / len(ngrams)
+            else:  # No ngrams of the word are present in self.ngrams
+                raise KeyError('all ngrams for word %s absent from model' % word)
+
+    def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_per=10000, update=False):
+        """ In word2vec, we built unigram dictionary, here we will make n-grams dictionary """
+
+        self.scan_vocab(sentences, progress_per=progress_per, trim_rule=trim_rule)  # initial survey
+        self.scale_vocab(keep_raw_vocab=keep_raw_vocab, trim_rule=trim_rule, update=update)  # trim by min_count & precalculate downsampling
+        self.finalize_vocab(update=update)  # build tables & arrays
+
+        # super(build_vocab, self, sentences, keep_raw_vocab=False, trim_rule=None, progress_per=10000, update=False)
+
+        self.init_ngrams()
+
+    def reset_ngram_weights(self):
+        for ngram in self.wv.ngrams:
+            # construct deterministic seed from word AND seed argument
+            self.wv.syn0_all[self.wv.ngrams[ngram]] = self.seeded_vector(ngram + str(self.seed))
+
+    def init_ngrams(self):
+        self.wv.ngrams = {}
+        self.wv.syn0_all = empty((self.bucket + len(self.wv.vocab), self.vector_size), dtype=REAL)
+        self.syn0_all_lockf = ones((self.bucket + len(self.wv.vocab), self.vector_size), dtype=REAL)  # zeros suppress learning
+
+        all_ngrams = []
+        for w, v in self.wv.vocab.items():
+            all_ngrams += Ft_Wrapper.compute_ngrams(w, self.min_n, self.max_n)
+        all_ngrams = set(all_ngrams)
+        self.num_ngram_vectors = len(all_ngrams)
+        logger.info("Total number of ngrams in the vocab is %d", self.num_ngram_vectors)
+
+        ngram_indices = []
+        for i, ngram in enumerate(all_ngrams):
+            ngram_hash = Ft_Wrapper.ft_hash(ngram)
+            ngram_indices.append(len(self.wv.vocab) + ngram_hash % self.bucket)
+            self.wv.ngrams[ngram] = i
+
+        self.wv.syn0_all = self.wv.syn0_all.take(ngram_indices, axis=0)
```

Comment:
This is not going to work if we want to resume training. Extraneous vectors should be discarded only when we are sure no training is going to take place further (probably analogous to `init_sims(replace=True)` in `Word2Vec`.